salt/cloud/clouds/nova.py

```diff
@@ -726,12 +726,18 @@ def request_instance(vm_=None, call=None):
 
         else:
             pool = floating_ip_conf.get('pool', 'public')
-            for fl_ip, opts in six.iteritems(conn.floating_ip_list()):
-                if opts['fixed_ip'] is None and opts['pool'] == pool:
-                    floating_ip = fl_ip
-                    break
-            if floating_ip is None:
+            try:
                 floating_ip = conn.floating_ip_create(pool)['ip']
+            except Exception:
+                log.info('A new ip address was unable to be allocated. '
+                         'An ip address will be pulled from the already allocated list, '
+                         'This will cause a race condition when building in parallel.')
```

Comment:
it won't fail.  There is just a race condition when building in parallel.

If you attempt to build with -P, each process will look up an ip in the address pool, and they will all get the same one, and then attempt to assign the same ip to each server.  Novaclient does not throw an error if this fails for some reason.

It is the same issue that this is fixing https://github.com/saltstack/salt/pull/42253 but #42253 was for the openstack driver, and this PR is for the nova driver.

If no floating ip address is able to be used, there is an exception thrown when `None` is passed to the floating_ip_associate function.

I can add a similar note about this race condition to the docs at the top of the nova driver?

```
+.. note::
 +
 +    When using floating ips from networks, if the Nova driver is unable to
 +    allocate a new ip address for the server, it will check that for
 +    unassociated ip addresses in the floating ip pool.  If SaltCloud is running
 +    in parallel mode, it is possible that more than one server will attempt to
 +    use the same ip address.
 +
```